### ISSUE TITLE
NTU Intel19 compiler error

### ISSUE THREAD

**Issue by davegill** [wrf-model/WRF#1444]
**Describe the bug**
```
ftn -o module_mp_ntu.o -c  -w -ftz -fno-alias -align all -FR -convert big_endian  -g -O0 -fno-inline -fno-ip -traceback -fpe0 -check-uninit -check noarg_temp_created,bounds,format,output_conversion,pointers,uninit -ftrapuv -unroll0 -u  -I../dyn_em -I../dyn_nmm  -I/scratch/wicker/ieva/WRF_test/WRF/external/esmf_time_f90  -I/scratch/wicker/ieva/WRF_test/WRF/main -I/scratch/wicker/ieva/WRF_test/WRF/external/io_netcdf -I/scratch/wicker/ieva/WRF_test/WRF/external/io_int -I/scratch/wicker/ieva/WRF_test/WRF/frame -I/scratch/wicker/ieva/WRF_test/WRF/share -I/scratch/wicker/ieva/WRF_test/WRF/phys -I/scratch/wicker/ieva/WRF_test/WRF/wrftladj -I/scratch/wicker/ieva/WRF_test/WRF/chem -I/scratch/wicker/ieva/WRF_test/WRF/inc -I/opt/cray/pe/netcdf/4.6.3.2/intel/19.0/include  -real-size `expr 8 \* 4` -i4  module_mp_ntu.f90
module_mp_ntu.f90(1617): error #5561: Subscript #2 of the array ZCCN has value 0 which is less than the lower bound of 1
            DUST_IN0 = DUST_IN0+ZCCN(IM,ID_DUST)
--------------------------------^
compilation aborted for module_mp_ntu.f90 (code 1)
```

**To Reproduce**
1. Use Intel19
2. Probably anything after this: dd7547db NTU microphysics scheme #1364

The problem is
```
      IF (ID_DUST*ID_IN.NE.0) THEN                                      ! IF DUST EXISTS
         DUST_IN0 = 0.
         DO IM = 1,NCCN
            DUST_IN0 = DUST_IN0+ZCCN(IM,ID_DUST)
         ENDDO
      ELSE                                                              ! NO DUST
         DUST_IN0 = BACKIN
      ENDIF
```

The issue is that ID_DUST == parameter, and when there is no dust, it equals zero, which is directly substituted.

I think the should be changed to:
```
      IF (ID_DUST*ID_IN.NE.0) THEN                                      ! IF DUST EXISTS
         DUST_IN0 = 0.
         ID_DUST_LOCAL = MAX( ID_DUST, 1 )
         DO IM = 1,NCCN
            DUST_IN0 = DUST_IN0+ZCCN(IM,ID_DUST_LOCAL)
         ENDDO
      ELSE                                                              ! NO DUST
         DUST_IN0 = BACKIN
      ENDIF
```

**Comment by davegill** [comment]
@louiswicker @weiwangncar @tzuchin12
This needs to be fixed

**Comment by davegill** [comment]
fixed by Corrections for ntu_init #1441
